test/e2e/network/service.go

```diff
@@ -1436,7 +1444,12 @@ var _ = SIGDescribe("Services", func() {
 			framework.Failf("Spec.ClusterIP was not cleared")
 		}
 		execPod := e2epod.CreateExecPodOrFail(cs, ns, "execpod", nil)
-		err = jig.CheckServiceReachability(externalNameService, execPod)
+		err = wait.PollImmediate(framework.Poll, CheckExternalNameServiceTimeOut, func() (done bool, err error) {
```

Comment:
I think that is better to add the `wait.PollImmediate`  here, so the method is reliable and people don´t have to add the active loops on top
https://github.com/kubernetes/kubernetes/blob/252887e39f905389156d2bc9c5932688857588e4/test/e2e/framework/service/jig.go#L986-L997

Reply:
right :), since `checkExternalServiceReachability` method only used by cases in service.go.